Rename and Reorganize

afrey15 opened the pull request:
## Number of Issue Fixed
- #72

### Changes Proposed in the Pull Request:
- Changed the names of all the resources and all the corresponding references in the code to snake_case
- Organized all the resources into folders, including a new organization for the Objects
- Cleaned up and organized all files in the repository
- Deleted files "label_description" and "Object_tree", which I do not believe are useful
- Moved "STATS" and "CONTRIBUTING" files to their rightful place in the wiki
- Moved other Github related files into the .github folder
- Attempted to rename the Issue Template folder and rename / move the Pull Request Template (tried to test but seems we'll need to merge to know for sure)
- Moved changelog to a different folder in the repository and turned it into a markdown file

### Known Problems

### Notes for Reviewer
- The "particles" folder in sprites and objects refers to intangible objects, that is, things you cannot physically touch or hold.

afrey15 commented on objects/obj_menu/Other_12.gml:
```diff
@@ -37,7 +37,7 @@ if (!global.online) {
         Local.Team = ds_list_find_value(teams, Local.player_id); //temporary
         Local.character = ds_list_find_value(classes, Local.player_id);
-        Local.sprite_index = scr_getSprite(Local.character);
+        Local.sprite_index = scr_get_character_sprite(Local.character);
```
Yes. That is why I made the change

afrey15 commented on sprites/spr_countdown/spr_countdown.yy:
```diff
@@ -1,96 +1,96 @@
     "gridX": 0,
     "gridY": 0,
-    "height": 16,
+    "height": 64,
```
I shrunk this one back down. This was another accidental delete :/

afrey15 commented:
I have a couple questions in here. I will do more testing tonight though to try to find more errors.

1. The idea here was that the How to Play section would be expanded and this would be a part of it. The page can be easily renamed if no one wants to expand on this though.
2. These changes were made to rename the Config folder so that it matched the style of the other folders. I did not intend for their to be so many commits but I had to figure out how to do it :/
3. ^
4. ^
5. ^
6. This event is now re-added
7. Every resource in GMS is stored in its own folder with the same name as the resource. That is why it looks like an unnecessary folder.
8. I don't remember doing this one. Everything appears to be in place still though..
9. The sprite's file name is changed when I checked the repository. Could you double check this one?
10. I mentioned changing the names of all resources, including sprites, but I will amend this if you are referring to something besides the resource names
11. The issue template names were already lowercase actually. The pull request template is the only one that was not lowercase.
12. I use the word "particles" here to refer to any object or sprite that is something that cannot be physically grabbed (not in the game, but if it existed in real life). Particles in GMS are intangible so that is why I chose the word. I mean to mention this in the notes so I will add it. Maybe "intangible" would be a better name?
13. ^
14. I placed these here because they deal with the physics of their respective items. The "level" folder is my most general Script folder if you prefer I move them there..
15. This event was re-added
16. I don't think this could have been caused by me. It is probably in master as well.. I can't find the code where coins are spawned.. Do you know where this is at?
17. The recharge station references are now changed
18. I will look into this one shortly

Regulator13 commented:
1. I don't think the stats belong on the how-to-play page no matter how big it is. Maybe a game information page or something.

2. Hold on, all that stuff was just to lowercase the 'C' on Config?  Even though that was the default given by gamemaker? That is something that should not have been messed with, especially since you can't be sure if a capital C is required. Unless you understand how gamemaker uses configs you shouldn't be messing with it just to make it match the others. Lots of things don't match, its not a big deal. I recommend you revert those commits.

7. I wasn't referring to the duplicate name, I was wondering why `obj_ball.yy` was being changed at all, I think like 10 lines were deleted

8. You see how changes that you don't remember making showing up is a problem, right?

9. `spr_HUD` is not changed to lowercase on your branch.

10. You make changes to actual sprites, not just the names, such as this file: /sprites/spr_countdown/05c4d29e-1ef0-440a-9460-882542ea6c95.png

11. The `ISSUE_TEMPLATE` folder should remain capitalized, the issue templates themselves we made so they can be whatever, the `PULL_REQUEST_TEMPLATE` should remain capitalized

12. But no one knows what you're referring too when you say particles. I don't think intangible is great either. Maybe bullets or something that describes them better

14. They really don't effect the physics of anything. I guess you can put them under level since they are used during the level

16. `scr_generate_platform`

18. Cool

afrey15 commented:
1. I'll make a new page for this
2. I researched configs for this. I do understand how they work.
3. obj_ball is one of the main objects that got deleted. It looks like it might be noting some residual information that got duplicated when I added back in the event. These are changes GMS made automatically to keep things working properly so I don't really want to mess with it
4. I see
5. It seems that GMS file organization is not case sensitive. That is why I was having so much trouble with the configs. I changed it again but HUD may or may not display as capital
6. This is now corrected
7. I am making these changes
8. Bullets doesn't really capture the category I'm trying to make. Can I post about it on Hangouts and then we will all be on the same page?
9. I mean, the one directly affects the physics of obj_ball. The other could be moved I guess.
10. Found it and corrected this

Regulator13 commented:
1. Cool
2. But why would you risk hundreds of filename changes just for a single lowercase letter change? You're taking making things look good too far.
3. Whenever you delete something (which I have no idea how you do accidently) you just gotta exit and discard changes. Snippets of code we don't understand is asking for problems. Normally I'd say throw out your pull request, but since you did 5000 lines of changes at once it'd take forever to do that. I guess we'll just have to trust that it still works.
5. What is `spr_huddy`?
7. Let's just not, the organization is fine as it is and even if it works I'd still want you to change it back to be organized logically.
8. I mean, you don't need my permission
9. Yeah but when I'm thinking of a script to use a gun I'm certainly not thinking about how it changes the physics, I'm thinking about how I'm using an item, which is the majority of the script. How about an item section?
10. You might want to remove that from your known problems then.

afrey15 commented:
1. ^
2. Okay, just to calm your fear of the unknown here. The configuration files are custom configurations you can use to automatically change what resources, objects, events, etc.. get included when you export to a different platform. It is similar to how our own game has custom config files that game has made (.ini). We are using the default config each time we compile the game and it has been working fine. I also tested created a zip file -- no problems. Nothing is at risk here. All that has been changed is the capital case of the top level folder and subsequently all the file extensions. If any problems ever occured with it, which they won't, it can be changed back just about instantly.
3. I was almost two hours into my work when it happened. I wasn't going to restart too easily on that. This wasn't like thinking out solutions and having it set either, it was just hundreds of small changes. I will do that in the future though for sure.
4. That sprite should not exist. Once again, the capital letters were too insignificant for GMS to recognize. I had to first change it to a different name to even get it to recognize the changes. It should be changed back.
5. I'd like to better understand how Github recognizes files. I believe it probably is the case that it needs to be all capital. I will test this on another repository though to put you at ease.
6. Cool
7. I will move them to the "level" folder. An items folder would cause a split in the scripts related to climbing, which I think should stay together if possible. I'm not sure what else would go in their either..
8. Okay

Regulator13 commented:
2. Okay but explaining what configs are doesn't actually make any difference here. This change might work but why make it at all? I had to review hundreds of file changes because you changed one letter. To remind you of your fear of the unknown you know very little about coding. What if we want to use another program that checks the config file using a script comparator? All the sudden the casing matters and it won't work for us because we changed from default. Furthermore the problem may not be obvious to what the problem is. I'm not saying this change will ruin everything. I'm saying changes like this are foolish. I know code fairly well and I know the more default you can keep things the better.
3. Commit your changes periodically. Seriously, deleting things is a big deal when you're working with a complex IDE
4. You added it in your last commit and have not changed it since.
5. Yeah don't test your stuff on our repository, make a test repository, I have one
6. Okay but some scripts are going to fall under two categories, do you have a plan for this or are you just gonna randomly sort them?
